rskj-core/src/main/java/co/rsk/peg/StandardMultisigFederation.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2017 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.peg;
+
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.bitcoinj.script.Script;
+import co.rsk.bitcoinj.script.ScriptBuilder;
+
+import java.time.Instant;
+import java.util.List;
+
+/**
+ * Immutable representation of an RSK Federation in the context of
+ * a specific BTC network.
+ *
+ * @author Ariel Mendelzon
+ */
+
+public class StandardMultisigFederation extends Federation {
+
+    public StandardMultisigFederation(
+        List<FederationMember> members,
+        Instant creationTime,
+        long creationBlockNumber,
+        NetworkParameters btcParams) {
+
+        super(members, creationTime, creationBlockNumber, btcParams);
+    }
+
+    @Override
+    public Script getRedeemScript() {
+        if (redeemScript == null) {
+            redeemScript = ScriptBuilder.createRedeemScript(getNumberOfSignaturesRequired(), getBtcPublicKeys());
+        }
+
+        return redeemScript;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) {
+            return true;
+        }
+
+        if (other == null || this.getClass() != other.getClass()) {
+            return false;
+        }
+
+        StandardMultisigFederation otherFederation = (StandardMultisigFederation) other;
+
+        return this.getNumberOfSignaturesRequired() == otherFederation.getNumberOfSignaturesRequired() &&
```

Comment:
We could also discuss this. What makes 2 federations be equal? If both have the same redeem script but different creation time, are they different? I think not, the way I see it the redeem script is basically what defines the federation.

Maybe same redeem script but different networks we could consider them different, at least the address will be different. What do you think?

Reply:
Im pretty sure that i agree on what you say. But is an interesting discussion to have and we should definitely have it.